src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/ReferenceCountedDisposable.cs

```diff
@@ -51,7 +53,7 @@ namespace Roslyn.Utilities
     /// even in concurrent execution.</para>
     /// </remarks>
     /// <typeparam name="T">The type of disposable object.</typeparam>
-    internal sealed class ReferenceCountedDisposable<T> : IReferenceCountedDisposable<T>, IDisposable
+    internal sealed class ReferenceCountedDisposable<T> : IReferenceCountedDisposable<T>
```

Comment:
IReferenceCountedDisposable is already IDisposable.  So i just removed this as being redundant.

Reply:
I felt it was clearer before. `ReferenceCountedDisposable<T>` and `IDisposable` are the primary use cases, and `IReferenceCountedDisposable<T>` is a secondary use case when none of the primary cases works. This change hides a primary interface behind a secondary one.

Reply:
ok.  reverted.